File: app/src/main/java/md/fusionworks/lifehack/ui/widget/TypefaceEditText.java
package md.fusionworks.lifehack.ui.widget;

import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.Typeface;
import android.util.AttributeSet;
import android.widget.EditText;

import md.fusionworks.lifehack.R;
import md.fusionworks.lifehack.util.TypefaceCache;

public class TypefaceEditText extends EditText {

  public TypefaceEditText(Context context) {
    super(context);
  }

  public TypefaceEditText(Context context, AttributeSet attrs) {
    super(context, attrs);

    TypedArray ta =
        context.getTheme().obtainStyledAttributes(attrs, R.styleable.TypefaceTextView, 0, 0);
    try {
      String font = ta.getString(R.styleable.TypefaceTextView_typeface);
      Typeface typeface = TypefaceCache.getTypeface(context, font);
      if (null != typeface) {
        setTypeface(typeface);
      }
    } finally {
      ta.recycle();
    }
  }

  public TypefaceEditText(Context context, AttributeSet attrs, int defStyle) {
    super(context, attrs, defStyle);

    TypedArray ta =
        context.getTheme().obtainStyledAttributes(attrs, R.styleable.TypefaceTextView, 0, 0);
    try {
      String font = ta.getString(R.styleable.TypefaceTextView_typeface);
      Typeface typeface = TypefaceCache.getTypeface(context, font);
      if (null != typeface) {
        setTypeface(typeface);
      }
    } finally {
      ta.recycle();
    }
  }
}

File: app/src/main/java/md/fusionworks/lifehack/ui/widget/TypefaceEditText.kt
package md.fusionworks.lifehack.ui.widget

import android.content.Context
import android.util.AttributeSet
import android.widget.EditText
import md.fusionworks.lifehack.R
import md.fusionworks.lifehack.util.TypefaceCache

class TypefaceEditText : EditText {

  constructor(context: Context) : super(context) {
  }

  constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {

    val ta = context.theme.obtainStyledAttributes(attrs, R.styleable.TypefaceTextView, 0, 0)
    try {
      val font = ta.getString(R.styleable.TypefaceTextView_typeface)
      val typeface = TypefaceCache.getTypeface(context, font)
      if (null != typeface) {
        setTypeface(typeface)
      }
    } finally {
      ta.recycle()
    }
  }

  constructor(context: Context, attrs: AttributeSet, defStyle: Int) : super(context, attrs,
      defStyle) {

    val ta = context.theme.obtainStyledAttributes(attrs, R.styleable.TypefaceTextView, 0, 0)
    try {
      val font = ta.getString(R.styleable.TypefaceTextView_typeface)
      val typeface = TypefaceCache.getTypeface(context, font)
      if (null != typeface) {
        setTypeface(typeface)
      }
    } finally {
      ta.recycle()
    }
  }
}